simulator/src/bin/generate-example-screenshots.rs

```diff
@@ -0,0 +1,211 @@
+
+        let path = format!("{}/{}.png", OUTPUT_BASE, $title);
+        $display
+            .to_image_buffer(&output_settings)
+            .save(&path)
+            .unwrap();
+
+        let screenshot = base64::encode(std::fs::read(&path).expect("Couldn't open screenshot"));
+
+        // Newlines in the code block aren't preserved by the stringify macro.
+        // Use {} in the code block to insert newlines in the generated output.
+        let doc_lines: Vec<_> = stringify!($code)
+                .trim_matches(|c| c == '{' || c == '}')
+                .trim()
+                .lines()
+                .map(|l| l.trim())
+                .map(|l| if l == "{ }" { "" } else { l }).collect();
+
+        // Note: empty lines must remain between HTML elements and inner Markdown for the Markdown
+        // to be parsed correctly.
+        println!(
+            r#"//! ## {}
+            //!
+            //! {}
+            //!
+            //! <div style="display: flex">
+            //! <img style="width: 128px; height: 128px; margin-right: 8px;" alt="{} example screenshot" src="data:image/png;base64,{}" />
+            //! <div style="flex-grow: 1;">
+//!
+//! ```rust
+//! # let mut display = embedded_graphics::mock_display::MockDisplay::default();
+//! {}
+//! # Ok::<(), core::convert::Infallible>(())
+//! ```
+//!
+//! </div>
+//! </div>
+//!"#,
+            $title,
+            $description.lines().collect::<Vec<_>>().join("\n//! "),
+            $title,
+            screenshot,
+            doc_lines.join("\n//! ")
+        );
+    };
+}
+
+fn main() {
+    let output_base = "./target/drawing-ops";
+    std::fs::create_dir_all(output_base).expect("Output directory could not be created");
+
+    let mut display: SimulatorDisplay<pixelcolor::Rgb888> =
+        SimulatorDisplay::new(Size::new(64, 64));
+
+    op!(
+        display,
+        "Draw a single pixel",
+        r#"This example draws a single green pixel.
+
+For cases where many pixels are drawn it is preferable to implement
+a custom iterator instead of calling `Pixel::draw` for each pixel, because
+some display drivers implement accelerated drawing of iterators."#,
+        {
+            use embedded_graphics::{pixelcolor::Rgb888, prelude::*};
+            {}
+            Pixel(Point::new(32, 32), Rgb888::GREEN).draw(&mut display)?;
+        }
+    );
+
+    op!(
+        display,
+        "Draw a line",
+        "This example draws a red line with 1px stroke.",
+        {
+            use embedded_graphics::{
+                pixelcolor::Rgb888, prelude::*, primitives::Line, style::PrimitiveStyleBuilder,
+            };
+            {}
+            Line::new(Point::new(16, 24), Point::new(48, 40))
+                .into_styled(
+                    PrimitiveStyleBuilder::new()
+                        .stroke_width(1)
+                        .stroke_color(Rgb888::RED)
+                        .build(),
+                )
```

Comment:
I missed this one.
```suggestion
                .into_styled(PrimitiveStyle::with_stroke(Rgb888::RED, 1))
```

Also, should we change this to something other than 1px width, now that the fabulous PR #271 is merged?

Reply:
Good idea. I've updated this example to use an 8px thick line. I moved the end point slightly to give the line a better look for the example's sake.